enhancements/agent-installer/interactive-network-config.md

```diff
@@ -0,0 +1,238 @@
+---
+title: interactive-network-config
+authors:
+  - "@zaneb"
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - "@andfasano"
+  - "@rwsu"
+  - "@bfournie"
+  - "@cgwalters"
+  - "@patrickdillon"
+approvers:
+  - "@celebdor"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - None
+creation-date: 2023-01-06
+last-updated: 2023-01-06
+tracking-link:
+  - https://issues.redhat.com/browse/AGENT-385
+see-also:
+  - "/enhancements/agent-installer/automated-workflow-for-agent-based-installer.md"
+replaces: N/A
+superseded-by: N/A
+---
+
+# Interactive Network Configuration
+
+## Summary
+
+Enable the user, after booting the agent ISO, to interactively add or correct
+any neccessary network configuration. Since we cannot rely on network
+connectivity for this step, configuration will be via a text user interface
+(TUI) displayed on the console.
+
+## Motivation
+
+Providing network configuration up-front, in the absence of the actual hosts to
+be used, is onerous for users. Names of NIC interfaces are difficult to predict
+in advance, and often MAC addresses are required. Even the simplified NMState
+format is difficult to write by hand.
+
+Correcting any error in the configuration passed into the installer currently
+requires returning to the beginning of the process, generating a new ISO and
+booting the hosts anew with the updated image. There is no guarantee of getting
+it right, so there could be multiple iterations of this process.
+
+### User Stories
+
+* As a cluster administrator, I want to install a cluster into a network
+  environment requiring custom network configuration (e.g. static IPs, NIC
+  bonds, VLANs) without needing any information from the physical servers in
+  advance.
+* As a cluster administrator, I want to fix a cluster installation that has
+  stalled due to incorrect network configuration without having to regenerate
+  the installation image.
+
+### Goals
+
+* Allow the user to configure static IPs, NIC bonds, and VLANs.
+* Make available all necessary information about the hardware and network
+  environemnt to do this configuration.
+* Provide an indication of when sufficient connectivity is or is not available.
```

Comment:
I think it could be useful, at this point of the doc, to provide a little bit more details about "sufficient connectivity", ie possibility for an host to reach the rendezvous host, to reach a registry, and so on (even though more info are provided later)